src/charts/line-chart/index.js

```diff
@@ -0,0 +1,385 @@
+      // oldest to newest
+      if (item.data.length > 0) {
+        if (moment.utc(item.data[0].timestamp).valueOf() - moment.utc(item.data[1].timestamp).valueOf() > 0) {
+          throw new Error([
+            `Line Chart data item index ${ct} ('${item.name}') looks to have data in the wrong order`,
+            `(item index 0 was newer than item index 1). Please presort data from oldest to newest`,
+            `before passing it to this chart.`
+          ].join(' '));
+        }
+      }
+
+      return item;
+    });
+    overlays = overlays || [];
+
+    if (typeof svgWidth !== 'number') {
+      throw new Error(`Line Chart is missing the property 'svgWidth'. This is required!`);
+    }
+    if (typeof svgHeight !== 'number') {
+      throw new Error(`Line Chart is missing the property 'svgHeight'. This is required!`);
+    }
+
+    /* xAxis */
+    /* yAxis */
+
+    /* xAxisStart can be undefined */
+    /* xAxisEnd can be undefined */
+    /* yAxisStart can be undefined */
+    /* yAxisEnd can be undefined */
+
+    overlayShowPoint = typeof overlayShowPoint === 'undefined' ? true : overlayShowPoint;
+    overlayPointRadius = typeof overlayPointRadius === 'undefined' ? 4.5 : overlayPointRadius;
+
+    // ------------------------------------------------------------------------
+    // ADJUST GRAPH SIZE
+    // ------------------------------------------------------------------------
+
+    const topMargin = 76;
+    const leftMargin = 56;
+    const rightMargin = 20;
+    const bottomMargin = 96;
+
+    const graphWidth = svgWidth - leftMargin - rightMargin;
+    const graphHeight = svgHeight - topMargin - bottomMargin;
+
+    svg.attr('width', svgWidth);
+    svg.attr('height', svgHeight);
+
+    layers.xAxis.attr('transform', `translate(${leftMargin},${topMargin+graphHeight})`)
+    layers.yAxis.attr('transform', `translate(${leftMargin},${topMargin})`)
+    layers.data.attr('transform', `translate(${leftMargin},${topMargin})`)
+
+    // ------------------------------------------------------------------------
+    // GATHERING DATA POINTS
+    // ------------------------------------------------------------------------
+    const dataPoints = {};
+
+    const defaultDataset = data.find(i => i.name === 'default');
+    if (defaultDataset.data.length === 0) {
+      return;
+    }
+
+    dataPoints.firstEvent = defaultDataset.data[0];
+    dataPoints.lastEvent = defaultDataset.data.slice(-1)[0];
+
+    dataPoints.firstEventXValue = moment.utc(dataPoints.firstEvent.timestamp).valueOf(); /* epoch ms */
+    dataPoints.firstEventYValue = dataPoints.firstEvent.value;
+    dataPoints.lastEventXValue = moment.utc(dataPoints.lastEvent.timestamp).valueOf(); /* epoch ms */
+    dataPoints.lastEventYValue = dataPoints.lastEvent.value;
+
+    dataPoints.eventWithSmallestValue = defaultDataset.data.reduce((a, b) => a.value < b.value ? a : b, {value: Infinity});
```

Comment:
Can we compile both the smallest and largest value by running through the data only once?

Reply:
Good point, I'll make that update